lib/opam_build.ml

```diff
@@ -95,15 +100,25 @@ let install_project_deps ~opam_version ~opam_files ~selection =
   @ pin_opam_files ~network groups @ [
     env "DEPS" non_root_pkgs;
     opam_depext;
+    unpin_opam_files ~network opam_files;
     run ~network ~cache "opam install $DEPS" ]
+  @ pin_opam_files ~network groups
+  @ [ run  ~network ~cache "opam install %s" root_pkgs ]
 
 let spec ~base ~opam_version ~opam_files ~selection =
   let open Obuilder_spec in
+  let build_each_package =
+    List.map
+        (fun x ->
+            Filename.basename x
+            |> Filename.chop_extension
+            |> run "opam exec -- dune build --only-packages=%s @install @check @runtest && rm -rf _build")
+        opam_files
```

Comment:
Doesn't this mean that we'd _always_ split multi-package repos, even when it's not necessary? That's going to slow down builds, as we'll lose parallelism?

Reply:
You are entirely correct. How can we detect the cases that need to run this way and the other ones? In a perfect solution, we would be able to see the interdependency between multiple repositories and run this way only when we fall into this case.

Reply:
Ok, it's more straightforward for me! Thanks for the clarification :)
I don't know if it is an issue, but how should the solver know that `happy-eyeballs-lwt` requires the local version of `happy-eyeballs` and not the one from `opam`?
I'm rewriting the solution without the caching issue order so we have the two solutions available.